packages/web3-eth/src/utils/prepare_transaction_for_signing.ts

```diff
@@ -64,19 +65,29 @@ const getEthereumjsTransactionOptions = (
 
 	let common;
 	if (!hasTransactionSigningOptions) {
-		common = Common.custom(
-			{
-				name: 'custom-network',
-				chainId: toNumber(transaction.chainId) as number,
-				networkId: !isNullish(transaction.networkId)
-					? (toNumber(transaction.networkId) as number)
-					: undefined,
-				defaultHardfork: transaction.hardfork ?? web3Context.defaultHardfork,
-			},
-			{
-				baseChain: web3Context.defaultChain,
-			},
-		);
+		// if defaultcommon is specified, use that.
+		if (web3Context.defaultCommon) {
+			common = web3Context.defaultCommon;
+
+			if (isNullish(common.hardfork))
+				common.hardfork = transaction.hardfork ?? web3Context.defaultHardfork;
+			if (isNullish(common.baseChain))
+				common.baseChain = web3Context.defaultChain as ValidChains;
+		} else {
+			common = Common.custom(
+				{
+					name: 'custom-network',
+					chainId: toNumber(transaction.chainId) as number,
+					networkId: !isNullish(transaction.networkId)
+						? (toNumber(transaction.networkId) as number)
+						: undefined,
+					defaultHardfork: transaction.hardfork ?? web3Context.defaultHardfork,
+				},
+				{
+					baseChain: web3Context.defaultChain,
+				},
+			);
```

Comment:
`hasTransactionSigningOptions` is false in this statement, which means that `transaction.chain`, `transaction.hardfork` and `transaction.common` is nullish. If `transaction.common` exists go to line 91 and use the `transaction.common` object.
We are using `custom-network` because in this case, `transaction.common.customChain.name` and `defaultCommon.customChain.name` does not exist.